### PR TITLE
Fix component style attribute parser

### PR DESCRIPTION
**Pull request by ianvieira**
When we have a style attribute on the original HTML, the parsing was not correct, specially with background-image with an external URL, this was not loading because of a split by ':' what breaks the URL. I added an library that generates a more complete style tag using css-to-react-native.

### CODE REVIEW

**Comment by remarkablemark** [comment]
Thanks for opening the pull request and bringing the bug to my attention @ianvieira

My question for you is do you think [css-to-react-native](https://github.com/styled-components/css-to-react-native) is too specific? Will it transform the styles correctly if rendered in the DOM? Or does it make sense to roll out a custom parser that is built on top of [css](https://github.com/reworkcss/css)?

Also, since `html-react-parser` is isomorphic, how should the parsing work in the browser? Let me know what you think.

**Comment by ianvieira** [comment]
@remarkablemark After using the new solution with the css-to-react-native, I've found some issues, so I tried another approach, that is more successful. Now I'm using AST to deconstruct the style string and then creating a new object with the values and property name in camel case. This new approach works faster as before and without the problem of using a isomorphic designed module.

**Comment by remarkablemark** [comment]
@ianvieira Nice! I like the AST approach because it'll support the general use cases.

The solution looks great but I found something unfortunate about `css-tree`. The browser library is [**165.7 kB**](https://unpkg.com/css-tree@1.0.0-alpha.26/dist/) minified which is huge.

The alternative `css` is [**10.81 kB**](https://unpkg.com/css@2.2.1/lib/parse/) unminified.

I have a library [`style-to-object`](https://github.com/remarkablemark/style-to-object) which is built on top of `css`. Could this be used instead? You might want to check out [iterator](https://github.com/remarkablemark/style-to-object#iterator).

I have a few more requests (if you would be so kind):
- Use [conventional commits](https://conventionalcommits.org/) (I'm trying to automate the CHANGELOG)
- Refactor `camelCase` in [`./lib/utilities.js`](https://github.com/remarkablemark/html-react-parser/blob/master/lib/utilities.js) with your method or [`camelize`](https://github.com/facebook/fbjs/blob/master/packages/fbjs/src/core/camelize.js)
  - _Note:_ There may be edge cases regarding [vendor prefixes](https://reactjs.org/docs/dom-elements.html#style)
- Write a test to ensure that inline styles with urls do not break

I want to thank you once more for looking into this and submitting the pull request. I really appreciate your time and effort. If you feel like you're swamped or have no time, I can always take over and lend a helping hand.

**Comment by ianvieira** [comment]
Of course, I can address these issues, in fact this will change will help me too, my application bundle is with 900kb and I'm trying to make it smaller. I'll do a new commit with the changes and also checks if your library handles my biggest problem, the background with a base64 image, which was not being interpreted rightly.

**Comment by ianvieira** [comment]
@remarkablemark I believe now it's how you expected. I tried to address all your requests and also added an base64 image for test, this way we're able to test this edge case and also the url.

File: lib/attributes-to-props.js [hunk 1]
```diff
@@ -5,6 +5,7 @@
  */
 var utilities = require('./utilities');
 var propertyConfig = require('./property-config');
+var parser = require('style-to-object');
```
Comment (remarkablemark) [inline lib/attributes-to-props.js]: Do you mind renaming `parser` to `styleToObject` just for keeping it explicit here?

File: lib/attributes-to-props.js [hunk 2]
```diff
@@ -63,33 +64,13 @@ function cssToJs(style) {
...
-    // css property itemized as key and value
-    var properties;
-    var j;
-    var propertiesLen;
+    var styleObj = {};
 
-    for (var i = 0, declarationsLen = declarations.length; i < declarationsLen; i++) {
-        properties = declarations[i].trim().split(':');
+    parser(style, function(propName, propValue) {
+        styleObj[utilities.camelCase(propName)] = propValue;
```
Comment (remarkablemark) [inline lib/attributes-to-props.js]: One thing about the style parser is that css comments is included in the iterator. Could we include an `undefined` check here?
```js
if (propName && propValue) {
  styleObj[utilities.camelCase(propName)] = propValue;
}
```

File: lib/utilities.js [hunk 3]
```diff
@@ -6,24 +6,19 @@
  * @param  {String} string - The string.
  * @return {String}
  */
+
+var _hyphenPattern = /-(.)/g;
```
Comment (remarkablemark) [inline lib/utilities.js]: Would you mind moving this line above the JSDoc? This is because we can find easier to read the comment when it's next to the function.

File: package.json [hunk 4]
```diff
@@ -30,7 +30,8 @@
   ],
   "dependencies": {
     "html-dom-parser": "0.1.2",
-    "react-dom-core": "0.0.2"
+    "react-dom-core": "0.0.2",
+    "style-to-object": "^0.2.0"
```
Comment (remarkablemark) [inline package.json]: Could you remove the `^`? I'm following `save-exact` since I don't have a `package-lock.json` or `yarn.lock` checked in.

**Comment by remarkablemark** [comment]
@ianvieira Thanks for addressing the requests so quickly. I made a few comments and then let's get this thing merged in.

Oh and before I forget, were you able to verify if the style prefix for `-ms` works as expected?

**Comment by ianvieira** [comment]
I’ll do the requested changes between today and tomorrow and yes I checked
the browser prefix and it’s working as expected, I also added this case to
the tests.

**Comment by ianvieira** [comment]
@remarkablemark I finished all the requested changes

**Comment by remarkablemark** [comment]
@ianvieira Awesome work! Also if you're looking to visualize your webpack bundle stats, check out [Webpack Visualizer](https://chrisbateman.github.io/webpack-visualizer/).

**Comment by ianvieira** [comment]
@remarkablemark Thank you! Great tip! I have another question, when this new version will be available at NPM? Currently I'm using my repo as source and I want to change to the package again

**Comment by remarkablemark** [comment]
@ianvieira I'll try to release either tonight or tomorrow morning.

**Comment by remarkablemark** [comment]
@ianvieira I have released `0.4.1`:
```sh
# npm
npm install html-react-parser@latest --save

# yarn
yarn upgrade html-react-parser --latest
```
